dripline/implementations/ethernet_scpi_provider.py

```diff
@@ -0,0 +1,229 @@
+import re
+import socket
+import threading
+
+import scarab
+
+from dripline.core import Service
+# Dripline Exceptions currently unavailable
+
+# logging currently unavailable
+# import logging
+# logger = logging.getLogger(__name__)
+
+__all__ = []
+
+
+__all__.append('EthernetSCPIProvider')
+class EthernetSCPIProvider(Service):
+    '''
+    A fairly generic subclass of Service for connecting to ethernet-capable instruments/devices.
+    In is developed for and tested against devices with a SCPI-compliant command set, but may be usable with devices which do not strictly conform.
+    In particular, devices must support sending a response to every command received (either natively, or via SCPI's command composition) and responses are expected to include a termination marking complete transmission.
+    '''
+    def __init__(self,
+                 socket_timeout=1.0,
+                 socket_info=('localhost', 1234),
+                 cmd_at_reconnect=['*OPC?'],
+                 reconnect_test='1',
+                 command_terminator='',
+                 response_terminator=None,
+                 bare_response_terminator=None, #TODO do we still need this? would a refactor be possible to put it elsewhere?
```

Comment:
this is only used for the Prologix repeaters to cover some weird cases

Reply:
ugh... I wonder if it makes more sense to clean things here and have a `PrlogixService(EthernetSCPIService)` to add deal with those edge cases, I think it makes this classes behavior harder to understand.